src/database/database.cpp

```diff
@@ -146,6 +174,95 @@ Database::~Database()
     Poco::Data::SQLite::Connector::unregisterConnector();
 }
 
+/***********************************************************************************************************************
+ * nodemanager::NodeInfoStorageItf implementation
+ **********************************************************************************************************************/
+
+aos::Error Database::SetNodeInfo(const aos::NodeInfo& info)
+{
+    try {
+        Poco::JSON::Object pocoNodeInfo;
+
+        const auto nodeInfo = Stringify(ConvertNodeInfoToJSON(info));
+        *mSession << "INSERT INTO nodeinfo (id, info) VALUES (?, ?);", bind(info.mID.CStr()), bind(nodeInfo), now;
+
+    } catch (const Poco::Exception& e) {
+        LOG_ERR() << "Failed to set node info: " << e.what();
+
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eFailed);
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+aos::Error Database::GetNodeInfo(const aos::String& nodeId, aos::NodeInfo& nodeInfo) const
+{
+    try {
+        Poco::Data::Statement statement {*mSession};
+
+        Poco::Nullable<std::string> pocoInfo;
+
+        statement << "SELECT info FROM nodeinfo WHERE id = ?;", bind(nodeId.CStr()), into(pocoInfo);
+        if (statement.execute() == 0) {
+            return aos::ErrorEnum::eNotFound;
+        }
+
+        nodeInfo.mID = nodeId;
+
+        if (!pocoInfo.isNull()) {
+            Poco::JSON::Parser parser;
+            const auto&        ptr = parser.parse(pocoInfo.value()).extract<Poco::JSON::Object::Ptr>();
+
+            AssertNotNull(ptr);
+            ConvertNodeInfoFromJSON(*ptr, nodeInfo);
+        }
+
+    } catch (const Poco::Exception& e) {
+        LOG_ERR() << "Failed to get node info: " << e.what();
+
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eFailed);
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+aos::Error Database::GetAllNodeIds(aos::Array<aos::StaticString<aos::cNodeIDLen>>& ids) const
+{
+    try {
+        Poco::Data::Statement    statement {*mSession};
+        std::vector<std::string> storedIds;
+
+        statement << "SELECT id FROM nodeinfo;", into(storedIds);
+        if (statement.execute() > ids.MaxSize()) {
```

Comment:
Consider checking PushBack result instead of this check

Reply:
done